Sources/momento/config/logging/LogProvider.swift

```diff
@@ -0,0 +1,17 @@
+class LogProvider {
+    internal static var momentoLoggerFactory: MomentoLoggerFactoryProtocol? = nil
+    
+    internal static func setLogger(loggerFactory: MomentoLoggerFactoryProtocol) {
+        self.momentoLoggerFactory = loggerFactory
+    }
+    
+    internal static func getLogger(name: String) throws -> MomentoLoggerProtocol {
+        if self.momentoLoggerFactory != nil {
+            return self.momentoLoggerFactory!.getLogger(loggerName: name)
+        } else {
+            throw FailedPreconditionError(
+                message: "Momento Logger Factory is not yet initialized"
+            )
+        }
+    }
+}
```

Comment:
Still a little unsure about this part. Figured it was best to initialize the LogProvider in the TopicClient using the configuration's logger factory, which would allow the user to specify the desired log level and logger class. But it makes getLogger have to check if it's been initialized yet and unsure if throwing an error here is the best response.

Reply:
Let's just provide the `DefaultMomentoLoggerFactory(level: .info)` as the default when we declare the var instead of making this nullable. That way users can override at client instantiation if desired and we get reasonable default performance without needing to throw errors.